### PR TITLE
Add Jenkins source from Faros AI to connector catalog

### PR DESCRIPTION
**Pull request by cjwooo**
## What
Adds the Jenkins source, developed and maintained by Faros AI, to the connector catalog. Test suites are for this source are run in https://github.com/faros-ai/airbyte-connectors.

## How
Jenkins source is implemented in Typescript.
Source code: https://github.com/faros-ai/airbyte-connectors/tree/main/sources/jenkins-source
Utilizes the faros-airbyte-cdk package, which is a Typescript version of the Python CDK: https://github.com/faros-ai/airbyte-connectors/tree/main/faros-airbyte-cdk

Successful source acceptance test: https://github.com/faros-ai/airbyte-connectors/runs/4157414583?check_suite_focus=true#step:8:317

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [ ] Documentation updated 
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/SUMMARY.md`
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: docs/integrations/sources/jenkins.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+AI](https://github.com/faros-ai/airbyte-connectors/tree/main/sources/jenkins-source)
+and supports both Full Refresh and Incremental syncs. You can choose if this
+connector will copy only the new or updated data, or all rows in the tables and
+columns you set up for replication, every time a sync is run.
+
+### Output schema
+
+Several output streams are available from this source:
+
+* [Builds](https://docs.gitlab.com/ee/api/commits.html) \(Incremental\)
```
Comment (cjwooo) [inline docs/integrations/sources/jenkins.md:15]: Can you provide proper API links here @Roma-Kyrnis

File: docs/integrations/sources/jenkins.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+AI](https://github.com/faros-ai/airbyte-connectors/tree/main/sources/jenkins-source)
+and supports both Full Refresh and Incremental syncs. You can choose if this
+connector will copy only the new or updated data, or all rows in the tables and
+columns you set up for replication, every time a sync is run.
+
+### Output schema
+
+Several output streams are available from this source:
+
+* [Builds](TODO) \(Incremental\)
```
Comment (cjwooo) [inline docs/integrations/sources/jenkins.md:15]: @Roma-Kyrnis Can you provide links to Jenkins API docs here?

File: docs/integrations/sources/jenkins.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+### Requirements
+
+* Jenkins Server
+* Jenkins User
+* Jenkins API Token
+
+### Setup guide
+
+TODO
```
Comment (cjwooo) [inline docs/integrations/sources/jenkins.md:47]: Can you provide instructions on how to generate the API token? @Roma-Kyrnis

**Comment by cjwooo** [comment]
@marcosmarxm @sherifnada I will add our remaining Typescript sources once we iron out the contribution process details with this one.

File: docs/integrations/sources/jenkins.md (hunk 2 above)
Comment (Roma-Kyrnis) [reply docs/integrations/sources/jenkins.md:15]: Ok @cjwooo I'll add

Comment (Roma-Kyrnis) [reply docs/integrations/sources/jenkins.md:15]: Can you update to?
Set `JOB_NAME` and `BUILD_NUMBER` to accept Jobs and Builds API schema
* [Builds]($JENKINS_SERVER_URL/job/$JOB_NAME/$BUILD_NUMBER/api/json?pretty=true) \(Incremental\)
* [Jobs]($JENKINS_SERVER_URL/job/$JOB_NAME/api/json?pretty=true)

**Comment by cjwooo** [comment]
@sherifnada I didn't add anything to `airbyte-integrations/builds.md` since the builds are run on our repo.

File: docs/SUMMARY.md [hunk 4]
```diff
@@ -70,6 +70,7 @@
     * [Instagram](integrations/sources/instagram.md)
     * [Intercom](integrations/sources/intercom.md)
     * [Iterable](integrations/sources/iterable.md)
+    * [Jenkins](integrations/sources/jenkins.md)
```
Comment (sherifnada) [inline docs/SUMMARY.md:73]: do you want to add something like `(sponsored by Faros.ai)`? up to you

File: docs/integrations/sources/jenkins.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
+# Jenkins
+
+## Overview
+
+The Jenkins source is maintained by [Faros
+AI](https://github.com/faros-ai/airbyte-connectors/tree/main/sources/jenkins-source)
```
Comment (sherifnada) [inline docs/integrations/sources/jenkins.md:6]:
```suggestion
AI](https://github.com/faros-ai/airbyte-connectors/tree/main/sources/jenkins-source). Please file any support requests on that repo to minimize response time from the maintainers. 
```

**Review by sherifnada — APPROVED** [review]
Amazing! thanks @cjwooo 

Is it intentional that this isn't shipping by default with Airbyte?

**Comment by cjwooo** [comment]
@sherifnada Not sure what you mean by "shipping by default with Airbyte?"

**Comment by sherifnada** [comment]
@cjwoo i mean available in the drop down for choosing connectors. To do that it needs to be added to the connector index like described in step 4 [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

**Comment by cjwooo** [comment]
@sherifnada I've updated the connector index. I just used an online UUID generator to get the UUID.

**Comment by marcosmarxm** [comment]
@sherifnada just for sanity I ran `./gradlew airbyte-config:init:processResources` locally and got the exception bellow:

```
Exception in thread "main" java.lang.RuntimeException: 
Failed to fetch valid spec file for docker image farosai/airbyte-jenkins-source:0.1.23 from GCS bucket io-airbyte-cloud-spec-cache.
 This will continue to fail until the connector change has been approved and published. 
See https://github.com/airbytehq/airbyte/tree/master/docs/connector-development#publishing-a-connector for more details.
```

Maybe we need to check if it's possible right now to add extenral connectors without breaking master building.

**Comment by sherifnada** [comment]
@marcosmarxm we'll need to publish the connector first

**Comment by marcosmarxm** [comment]
Depends on https://github.com/airbytehq/airbyte/issues/8138 to be published.

**Pull request merged**